envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt

```diff
@@ -120,6 +121,10 @@ internal class EnvoyEgressRoutesFactory(
             routeAction.setInternalRedirectAction(RouteAction.InternalRedirectAction.HANDLE_INTERNAL_REDIRECT)
         }
 
+        routeAction
+            .setAutoHostRewrite(BoolValue.of(true))
+            .setAutoHostRewriteHeader("Custom-Host")
```

Comment:
Shouldn't this option be somehow whitelisted? Any client can rewrite Host header using this mechanism.